src/libsyntax/tokenstream.rs

```diff
@@ -368,8 +368,21 @@ impl TokenStream {
     // This is otherwise the same as `eq_unspanned`, only recursing with a
     // different method.
     pub fn probably_equal_for_proc_macro(&self, other: &TokenStream) -> bool {
-        let mut t1 = self.trees();
-        let mut t2 = other.trees();
+        fn semantic_tree(tree: &TokenTree) -> bool {
+            match tree {
+                | TokenTree::Token(_, Token::Comma)
+                | TokenTree::Token(_, Token::OpenDelim(DelimToken::NoDelim))
+                | TokenTree::Token(_, Token::CloseDelim(DelimToken::NoDelim))
+                | TokenTree::Token(_, Token::OpenDelim(DelimToken::Brace))
+                | TokenTree::Token(_, Token::CloseDelim(DelimToken::Brace))
+                | TokenTree::Token(_, Token::Semi)
```

Comment:
Like above, could a comment be added here why this isn't necessary to compare?

I'm also a little uneasy about this one like commas though because it seems like we're saying that semicolons are optional in Rust, but I feel like there's something that can parse one way or another with and without a semicolon

Reply:
I agree, but the thing to consider here is whether there's going to be a modification to the AST that add/removes semicolons without preserving semantics. I'd be surprised if that was the case.